include/boost/geometry/strategies/cartesian/densify.hpp

```diff
@@ -0,0 +1,121 @@
+// Boost.Geometry
+
+// Copyright (c) 2017-2018, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_CARTESIAN_DENSIFY_HPP
+#define BOOST_GEOMETRY_STRATEGIES_CARTESIAN_DENSIFY_HPP
+
+
+#include <boost/geometry/algorithms/detail/convert_point_to_point.hpp>
+#include <boost/geometry/algorithms/detail/signed_size_type.hpp>
+#include <boost/geometry/arithmetic/arithmetic.hpp>
+#include <boost/geometry/arithmetic/dot_product.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/strategies/densify.hpp>
+#include <boost/geometry/util/math.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace densify
+{
+
+
+template
+<
+    typename CalculationType = void
+>
+struct cartesian
+{
+    template <typename Point, typename AssignPolicy, typename T>
+    static inline void apply(Point const& p0, Point const& p1, AssignPolicy & policy, T const& length_threshold)
+    {
+        typedef typename AssignPolicy::point_type out_point_t;
+        typedef typename select_most_precise
+            <
+                typename coordinate_type<Point>::type,
+                typename coordinate_type<out_point_t>::type,
+                CalculationType
+            >::type calc_t;
+
+        typedef model::point<calc_t, 2, cs::cartesian> point2d_t;
+        
+        point2d_t const xy0(geometry::get<0>(p0), geometry::get<1>(p0));
+        point2d_t const xy1(geometry::get<0>(p1), geometry::get<1>(p1));
+        // dir01 = xy1 - xy0
+        point2d_t dir01 = xy1;
+        geometry::subtract_point(dir01, xy0);
+        calc_t const dot01 = geometry::dot_product(dir01, dir01);
+        calc_t const len2d = math::sqrt(dot01);
+
+        // TODO: For consistency with spherical and geographic 2d length is
+        // taken into account. This probably should be changed. Also in the
+        // other strategies dimensions > 2 should be taken into account.
+        signed_size_type n = signed_size_type(len2d / length_threshold);
+        if (n <= 0)
+            return;
+
+        // NOTE: Normalization will not work for integral coordinates
+        // normalize
+        //geometry::divide_value(dir01, len2d);
```

Comment:
No, there is no need for that. Below equivalent formula is used. It's requires more operations since vector multiplication and division is done each iteration. An alternative would be to use it only for integral coordinates and for FP or user-defined coordinates used normalized vector. But I'm not sure if it's that important. What do you think?

Reply:
No, I thought integer was not supported at all. Now I understand, thanks. So you can ignore my comment.